### PR TITLE
Define Monoid instance for ReaderT

### PR DESCRIPTION
**Pull request by safareli**

### CODE REVIEW

**Comment by safareli** [comment]
[here](https://github.com/slamdata/purescript-formatters/pull/17/commits/b304aa406c3f79265dc1580f22a48d5df00d8b50) is Monoid instance in action

**Comment by safareli** [comment]
ping

File: src/Control/Monad/Reader/Trans.purs [hunk 1]
```diff
@@ -72,6 +74,12 @@ instance monadReaderT :: Monad m => Monad (ReaderT r m)
 
 instance monadZeroReaderT :: MonadZero m => MonadZero (ReaderT r m)
 
+instance semigroupReaderT :: (Monad m, Semigroup a) => Semigroup (ReaderT s m a) where
```
Comment (paf31) [inline src/Control/Monad/Reader/Trans.purs]: only need `Apply` here, not a full `Monad`.

File: src/Control/Monad/Reader/Trans.purs [hunk 2]
```diff
@@ -72,6 +74,12 @@ instance monadReaderT :: Monad m => Monad (ReaderT r m)
...
+  append = lift2 (<>)
+
+instance monoidReaderT :: (Monad m, Monoid a) => Monoid (ReaderT s m a) where
```
Comment (paf31) [inline src/Control/Monad/Reader/Trans.purs]: only needs `Applicative`.

File: src/Control/Monad/Reader/Trans.purs [hunk 3]
```diff
@@ -20,6 +20,8 @@ import Control.Monad.Trans.Class (class MonadTrans, lift)
 import Control.Monad.Writer.Class (class MonadWriter, class MonadTell, pass, listen, tell)
 import Control.MonadPlus (class MonadPlus)
 import Control.MonadZero (class MonadZero)
+import Data.Monoid (class Monoid, mempty)
```
Comment (kRITZCREEK) [inline src/Control/Monad/Reader/Trans.purs]: Please keep the sorting

**Comment by paf31** [comment]
Thanks!

**Pull request merged**